### PR TITLE
RX65N DTC Driver Support Added

### PR DESCRIPTION
**Pull request by Bhindhiya**
## Summary

RX65N DTC driver support added

## Impact

Porting the RX65N DTC driver

### CODE REVIEW

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 1]
```diff
@@ -0,0 +1,1822 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:35]: Fix License to Apache

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 2]
```diff
@@ -0,0 +1,1822 @@
...
+
+#include "chip.h"
+#include "up_arch.h"
+#include "rx65n_definitions.h"
+#include "rx65n_dtc.h"
+
+#include <arch/board/board.h>
+#include <arch/board/rx65n_gpio.h>
+
+#if defined (CONFIG_RX65N_DTC)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:64]: There shouldn't have space after "defined"

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 3]
```diff
@@ -0,0 +1,1822 @@
...
+
+/* DTC IP version */
+
+#define DTC_IP_VER_DTC      (0)
+#define DTC_IP_VER_DTCa     (1)
+#define DTC_IP_VER_DTCb     (2)
+
+/* Supportted DTC IP version and channel number */
+
+#if defined(CONFIG_ARCH_BOARD_RX65N_RSK2MB) || defined( CONFIG_ARCH_BOARD_RX65N_GRROSE)
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:78]: Shouldn't have space after "defined("

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 4]
```diff
@@ -0,0 +1,1822 @@
...
+void rx65n_dtc_srcactivation(DTC_HANDLE handle, uint8_t src)
+{
+  struct rx65n_dtc_s *dtchandle = (struct rx65n_dtc_s *)handle;
+
+  DEBUGASSERT(handle != NULL);
+
+  dtc_activation_source_t act_source = (dtc_activation_source_t)src;
+
+  DEBUGASSERT(act_source >= DTC_MIN_VECT_NUMBER
+                  && act_source <= DTC_MAX_VECT_NUMBER);
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1232]: move && to previous line and align "act_source" below "act_source"

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 5]
```diff
@@ -0,0 +1,1822 @@
...
+                                          uint8_t src, uint32_t dcfg,
+                                uint32_t nseq_transfer, uint8_t nseq)
+{
+  struct rx65n_dtc_s *dtchandle = (struct rx65n_dtc_s *)handle;
+  dtc_err_t ret = DTC_SUCCESS;
+  uint32_t *ptr         = NULL;
+  dtc_transfer_data_t *p_transfer_data = NULL;
+  uint8_t   rrs_backup  = 0;
+  uint8_t   dtce_backup = 0;
+
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1305]: no reason for this empty line here

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 6]
```diff
@@ -0,0 +1,1822 @@
...
+  int i;
+
+  /* Transfer Information count */
+
+  uint32_t count = nseq_transfer;
+
+  /* Activation source */
+
+  dtc_activation_source_t act_source = (dtc_activation_source_t)src;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1422]: act_source should be declared at the beginning of the function

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 7]
```diff
@@ -0,0 +1,1822 @@
...
+  if ((act_source < DTC_MIN_VECT_NUMBER)
+           || (act_source > DTC_MAX_VECT_NUMBER))
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1424]: move || to previous line and align (act_source with (act_source

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 8]
```diff
@@ -0,0 +1,1822 @@
...
+    {
+      return DTC_ERR_INVALID_ARG;
+    }
+
+  /* Static User Configuration */
+
+  dtc_static_transfer_data_cfg_t *p_stransfer_cfg =
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1431]: declare p_stransfer_cfg at in the top

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 9]
```diff
@@ -0,0 +1,1822 @@
...
+                      (dtc_static_transfer_data_cfg_t *)scfg;
+  if (p_stransfer_cfg == NULL)
+    {
+      return DTC_ERR_INVALID_ARG;
+    }
+
+  /* Allocate  memory in RAM for transfer information */
+
+  dtc_transfer_data_t *p_transfer_data = (dtc_transfer_data_t *)pdt;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1440]: ditto

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 10]
```diff
@@ -0,0 +1,1822 @@
...
+                                                uint8_t src,
+                                      uint32_t dcfg, uint32_t nchain)
+{
+  struct rx65n_dtc_s *dtchandle = (struct rx65n_dtc_s *)handle;
+  dtc_err_t ret = DTC_SUCCESS;
+  uint32_t *ptr         = NULL;
+  dtc_transfer_data_t *p_transfer_data = NULL;
+  uint8_t   rrs_backup  = 0;
+  uint8_t   dtce_backup = 0;
+
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1515]: no empty line

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 11]
```diff
@@ -0,0 +1,1822 @@
...
+  uint32_t count = nchain + 1;
+
+  /* Dynamic user configuration */
+
+  dtc_dynamic_transfer_data_cfg_t *p_dtransfer_cfg =
+                           (dtc_dynamic_transfer_data_cfg_t *)dcfg;
+
+  /* Activation source */
+
+  dtc_activation_source_t act_source = (dtc_activation_source_t)src;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1530]: ditto

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 12]
```diff
@@ -0,0 +1,1822 @@
...
+
+  int i;
+
+  /* Transfer Information count */
+
+  uint32_t count = nchain + 1;
+
+  /* Activation source */
+
+  dtc_activation_source_t act_source = (dtc_activation_source_t)src;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1619]: ditto

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 13]
```diff
@@ -0,0 +1,1822 @@
...
+  if ((act_source < DTC_MIN_VECT_NUMBER)
+              || (act_source > DTC_MAX_VECT_NUMBER))
+    {
+      return DTC_ERR_INVALID_ARG;
+    }
+
+  /* Static User Configuration */
+
+  dtc_static_transfer_data_cfg_t *p_stransfer_cfg =
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1628]: ditto

File: arch/renesas/src/rx65n/rx65n_dtc.c [hunk 14]
```diff
@@ -0,0 +1,1822 @@
...
+                          (dtc_static_transfer_data_cfg_t *)scfg;
+  if (p_stransfer_cfg == NULL)
+    {
+      return DTC_ERR_INVALID_ARG;
+    }
+
+  /* Allocate  memory in RAM for transfer information */
+
+  dtc_transfer_data_t *p_transfer_data = (dtc_transfer_data_t *)pdt;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_dtc.c:1637]: ditto

**Review by acassis — CHANGES_REQUESTED** [review]

File: arch/renesas/src/rx65n/rx65n_dtc.c (hunk 14 above)
Comment (Bhindhiya) [reply arch/renesas/src/rx65n/rx65n_dtc.c]: Thank you for the comments.

The comments have been addressed in b0ae3fe.

**Pull request merged**